### ISSUE TITLE
Upgrade publish workflow to zip contents

### ISSUE THREAD

**Issue by Noggog** [Starfield-Community-Patch/Starfield-Community-Patch#246]
Right now it just publishes the Spriggit converted esm file directly to the Github release.

Eventually we might have other files such as:
- Loose readmes
- ini files
- Assets to be included loosely
- Assets to be packed in a BSA

So, this release will eventually need to be something like a zip instead.

Additionally, we'll need to figure out the file structure a bit:
```
src/
   plugin/                                Spriggit files for the esm
       Npcs/
           SomeNpc.yaml
   external/                              Loose files
      Docs/
         readme.txt
   assets/                                Assets to be packed in the BSA
      meshes/
         some-mesh.nif
```

Where the packaging workflow would go to these folders and:
- Run the spriggit translation on `plugin/`
- Run any BSA packing (TBD) on `assets/`
- Include any files in `external/`

All into one zip to be published:
```
StarfieldCommunityPatch.esm
StarfieldCommunityPatch.ba2
Docs/
   readme.txt
```

The BSA bits aren't sorted out yet, so this card would just be including `external/` and doing the zipping, for now

**Comment by Noggog** [comment]
Liking the names of 
- `packed` instead of `assets`
- `loose` instead of `external`

So could go with that naming instead, if we think it's clearer

**Comment by Pickysaurus** [comment]
I am against using a `src` subfolder as it makes this project harder to work with inside the Creation Kit. 

My feeling is that all data should be relative to the game's "Data" folder with a dedicated subfolder for Spriggit data. When running the build (locally) the resulting BA2/ESM/INI/etc can post to a "packed" folder and maybe the final ZIP too. 

Doing it this way means that contributors can Git Clone the repo into a MO2 mod folder, build the current plugin and start editing.

**Comment by Noggog** [comment]
Good thoughts.   Tinkering around with it so far it's got some complications:
- Harder to distinguish between what is repo "litter" like `Contributing.md`, `.gitattributes`, etc.   Any publishing scripts will need to be made smarter.
- For example, the workflow we have set up looks for changes in `/src` as a way to only publish if actual changes were made.  This would no longer work so easily.  Maybe not the biggest deal.
- Flattening to top level, paths like `src/loose/Docs` would now just be `Docs/`, and again be confusing with the other repo litter.  Is it the repo's docs?  or the docs to be packed?
![image](https://github.com/Starfield-Community-Patch/Starfield-Community-Patch/assets/24981326/aaae7b42-41f4-497f-a434-360f37ec5158)

However, I do like the idea of just being able to clone into Mo2 and start the game, and being able to edit meshes directly and have them already be in the correct spot for commit/upload.  That's definitely appealing

**Comment by Noggog** [comment]
closed with #277
